projects/angular/src/layout/nav/aria-current-link.spec.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2016-2022 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Component } from '@angular/core';
+import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { Router } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+
+import { MainContainerWillyWonka } from './chocolate/main-container-willy-wonka';
+import { ClrNavigationModule } from './navigation.module';
+import { ClrAriaCurrentLink } from './aria-current-link';
+@Component({
+  template: `
+    <clr-header class="header">
+      <div class="header-nav">
+        <a clrAriaCurrentLink class="nav-link" routerLink="/home" routerLinkActive="active"
+          ><span class="nav-text">Components</span></a
+        >
+        <a clrAriaCurrentLink class="nav-link" routerLink="/patterns" routerLinkActive="active"
+          ><span class="nav-text">Patterns</span></a
+        >
+      </div>
+    </clr-header>
+  `,
+})
+class TestComponent {}
+
+describe('AriaCurrentLink Directive', () => {
+  let fixture: ComponentFixture<any>;
+  let router: Router;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        ClrNavigationModule,
+        RouterTestingModule.withRoutes([
+          { path: 'home', component: TestComponent },
+          { path: 'patterns', component: TestComponent },
+        ]),
+      ],
+      declarations: [TestComponent, ClrAriaCurrentLink],
+      providers: [MainContainerWillyWonka],
+    });
+
+    fixture = TestBed.createComponent(TestComponent);
+    router = TestBed.inject(Router);
+    fixture.detectChanges();
+  });
+
+  afterEach(() => {
+    fixture.destroy();
+  });
+
+  it('directive selects links', () => {
+    const linkElements = fixture.debugElement.queryAll(By.directive(ClrAriaCurrentLink));
+    expect(linkElements.length).toBe(2);
+  });
+
+  it('decorates the active link with aria-current', fakeAsync(() => {
+    router.navigate(['/home']);
+    tick();
+    fixture.detectChanges();
+    let activeLink = fixture.debugElement.query(By.css('.active'));
+    expect(activeLink.nativeElement.href).toMatch(/home/);
+    expect(activeLink.nativeElement.getAttribute('aria-current')).toEqual('page');
```

Comment:
would we want to make sure here and in the block below that the other link doesn't have `aria-current` on it?

Reply:
I assumed using `query` instead of `queryAll` would throw an exception if more than one element was returned, but in hindsight that's checking the implementation too much (that aria-current relies on the active css class) instead of the specific behavior we want.

Updated

Reply:
Maybe give the two links unique classnames so that you can query them outside of them being active?

This is largely just to make sure that `aria-current` is unset as expected with the active link is changed.

Reply:
Makes sense - updated.